spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -383,14 +390,31 @@ public boolean dropNamespace(String[] namespace) throws NoSuchNamespaceException
 
   @Override
   public final void initialize(String name, CaseInsensitiveStringMap options) {
-    this.cacheEnabled = Boolean.parseBoolean(options.getOrDefault("cache-enabled", "true"));
+    this.cacheEnabled = PropertyUtil.propertyAsBoolean(options,
+        CatalogProperties.TABLE_CACHE_ENABLED, CatalogProperties.TABLE_CACHE_ENABLED_DEFAULT);
+
+    // If the user disabled caching, we turn off the cache-expiration for them
+    long defaultCacheExpirationInterval = cacheEnabled ?
+            CatalogProperties.TABLE_CACHE_EXPIRATION_INTERVAL_MS_DEFAULT :
+            CatalogProperties.TABLE_CACHE_EXPIRATION_INTERVAL_MS_OFF;
+
+    this.cacheExpirationIntervalMs = PropertyUtil.propertyAsLong(options,
+        CatalogProperties.TABLE_CACHE_EXPIRATION_INTERVAL_MS,
+        defaultCacheExpirationInterval);
```

Comment:
I think this should use `CatalogProperties.TABLE_CACHE_EXPIRATION_INTERVAL_MS_DEFAULT` instead of setting the default based on `cacheEnabled`. If the cache is disabled, then the expiration interval is ignored. There is no need for extra logic to set it to some value when we will ignore it.

Reply:
Sure. That makes sense.

Reply:
Done.